src/new/wallet.js

```diff
@@ -2211,11 +2211,14 @@ class HathorWallet extends EventEmitter {
    *
    * @param {string} address Address to get the index
    *
-   * @return {Promise<number>}
+   * @return {Promise<number | null>}
    **/
   async getAddressIndex(address) {
-    const addresInfo = await this.storage.getAddressInfo(address);
-    return addresInfo.bip32AddressIndex;
+    const addressInfo = await this.storage.getAddressInfo(address);
+    if (!addressInfo) {
+      return null;
+    }
+    return addressInfo.bip32AddressIndex;
```

Comment:
suggestion(if-minor)

```suggestion
    return addressInfo?.bip32AddressIndex ?? null;
```

The [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is just to safely access the index, but it returns `undefined` so the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) can return null (and it is safe for zero value unline `||`).

Reply:
I ended up doing @andreabadesso's suggestion (https://github.com/HathorNetwork/hathor-wallet-lib/pull/598/commits/210c596022cfe1c02fd375fea05add2c4dd29277) which is pretty similar. And added a test for this method which was missing